deepspeed/runtime/engine.py

```diff
@@ -219,6 +219,8 @@ def __init__(
         self.collate_fn = collate_fn
         self.mpu = mpu
         self.data_parallel_group = None
+        self.shard_parallel_group = None
```

Comment:
I don't think this should exist in engine since it is zero-specific.

Reply:
Another way of saying this is that I don't see how these will be used in engine.